libpod/container.go

```diff
@@ -808,3 +814,42 @@ func (c *Container) isStopped() (bool, error) {
 	}
 	return c.state.State == ContainerStateStopped, nil
 }
+
+// nil container state to the database
+func (c *Container) save() error {
+	if err := c.runtime.state.SaveContainer(c); err != nil {
+		return errors.Wrapf(err, "error saving container %s state", c.ID())
+	}
+	return nil
+}
+
+// CleanupStorage unmounts all mount points in container and cleans up container storage
+func (c *Container) CleanupStorage() error {
+	for _, mount := range c.mounts {
+		if err := unix.Unmount(mount, unix.MNT_DETACH); err != nil {
+			logrus.Warnf("container %s failed to unmount %s : %v", c.ID(), mount, err)
+		}
+	}
+	c.mounts = []string{}
+
+	if c.state.Mounted {
+		// TODO Remove this once --detached containers are handled
+		// Attempt to umount shm tmpfs mounted in the bundlePath ignoring error.
+		// This is a bit of a hack since we have no way of cleaning this up with --detached containers at this point
+		shmdir := filepath.Join(c.bundlePath(), "shm")
+		_ = unix.Unmount(shmdir, unix.MNT_DETACH)
```

Comment:
Instead of ignoring errors here, log them with logrus so we have some idea things are going wrong

Reply:
Ok I will ignore syscall.EINVAL (Not mounted) and syscal.ENOENT (Directory does not exist) both expected errors.